[#11010] Show students created/updated in-place in instructor enrollment page table

ypinhsuan opened the pull request:
Fixes #11010

**Outline of Solution**
Highlight the rows in the enroll table to reflect the status. A color legend is added below the table. 
The colors used are the same as the summary table below

![ezgif com-gif-maker](https://user-images.githubusercontent.com/58677983/122527659-1bf50100-d04e-11eb-9ad1-cfe07546e59d.gif)

damithc commented:
For the spreadsheet, is it possible to use lighter shades of the colors? but OK to use dark shades if it's harder to use lighter shades. For added students, can use green instead? 'no changes' can remain white or use light grey?

ypinhsuan commented:
I have used lighter colors for the table which looks better. Unmodified students will remain white.
![image](https://user-images.githubusercontent.com/58677983/122540525-5022ee80-d05b-11eb-8df8-f511aa677df2.png)

damithc commented:
> I have used lighter colors for the table which looks better. Unmodified students will remain white.
> ![image](https://user-images.githubusercontent.com/58677983/122540525-5022ee80-d05b-11eb-8df8-f511aa677df2.png)

Looks good to me @ypinhsuan 

For consistency, we might want to change the panel below to follow the darker shades of the same colors. In particular, for 'added students'.

ypinhsuan commented:
Changed the panel color for new students to green and color for students updated with no changes to grey
![image](https://user-images.githubusercontent.com/58677983/122547107-3fc24200-d062-11eb-844d-9d94dd1798e7.png)

damithc commented:
> Changed the panel color for new students to green and color for students updated with no changes to grey
> ![image](https://user-images.githubusercontent.com/58677983/122547107-3fc24200-d062-11eb-844d-9d94dd1798e7.png)

Looks good. For the panel 'no changes', can go with the default style (see below), which is closer to white.

![image](https://user-images.githubusercontent.com/1673303/122550144-eb20c600-d065-11eb-90ff-8f3bdba2bdf7.png)

On a separate note, perhaps the errors panel should be on top as that's the one we don't want the user to miss?

ypinhsuan commented:
That means there will two panels with the same color (unmodified and not changed)? Or would it better to combine the 2 panels into one?

damithc commented:
> That means there will two panels with the same color (unmodified and not changed)? Or would it better to combine the 2 panels into one?

Oh, I didn't realize there are two. OK to keep as two. In that case, use light grey in the spreadsheet to match the panel color?

ypinhsuan commented:
Changed the color for students updated with no changes in spreadsheet to grey and updated the order of summary panel.
error -> new students -> modified students -> students updated with no changes -> unmodified students
![1](https://user-images.githubusercontent.com/58677983/122629164-a3458180-d0ed-11eb-80ec-752c1acb1758.gif)

damithc commented:
> Changed the color for students updated with no changes in spreadsheet to grey and updated the order of summary panel.
> error -> new students -> modified students -> students updated with no changes -> unmodified students

UI looks good to me me @ypinhsuan 👍

halfwhole commented on src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html:
```diff
@@ -108,6 +108,12 @@ <h1 id="enroll-header">Enroll Students for {{ courseId }}</h1>
+              <div class="box" style="background-color: #F19491;"></div><span class="mr-5">Student failed to be enrolled</span>
+              <div class="box" style="background-color: #AEEBA6;"></div><span class="mr-5">Student added</span>
+              <div class="box" style="background-color: #FCF4B3;"></div><span class="mr-5">Student modified</span>
+              <div class="box" style="background-color: #DDD;"></div><span class="mr-5">Student updated with no changes</span>
```
Could there be a way to extract these colours here along with those in styles into the same place?

halfwhole reviewed (approved):
LGTM

ccyccyccy reviewed (commented):
Good job overall. Just some minor comments

ypinhsuan commented on src/e2e/java/teammates/e2e/pageobjects/InstructorCourseEnrollPage.java:
```diff
@@ -113,6 +113,9 @@ public void verifyResultsPanelContains(StudentAttributes[] expectedNewStudents,
         // number of tables depends on what results are present
         int numTables = 0;
         List<WebElement> tables = resultsPanel.findElements(By.tagName("table"));
+        if (expectedErrorStudents != null) {
```
The order of the tables was changed and now the error table is now the first table. Error students need to be checked first because of the way the tables are retrieved by `numTables`.

ccyccyccy reviewed (approved):
LGTM!
